pkgs/development/python-modules/torch-scatter/default.nix

```diff
@@ -0,0 +1,31 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  setuptools,
+  torch,
+}:
+
+buildPythonPackage rec {
+  pname = "torch-scatter";
+  version = "2.1.2";
+  src = fetchFromGitHub {
+    owner = "rusty1s";
+    repo = "pytorch_scatter";
+    tag = version;
+    hash = "sha256-dmJrsWoFsqFlrgfbFHeD5f//qUg0elmksIZG8vXXShc=";
+  };
+  pyproject = true;
+
+  build-system = [ setuptools ];
+
+  dependencies = [ torch ];
+
+  pythonImportsCheck = [ "torch_scatter" ];
+
+  meta = with lib; {
```

Comment:
remove with lib; as it increases eval times

Reply:
why is this? is there a benchmark comparison?

Reply:
Yes. Don't remember where to find it. Basically, when you use `with`, you put all of its children into the eval space. `lib` has a lot of child functions and values. This uses more ram and more cpu as Nix has to parse more and check against what is loaded into context

Reply:
done in 110b0aa09b26c012c18e008680ad406151e57a11